site/content/docs/04-preloading.md

```diff
@@ -132,3 +132,23 @@ You can abort rendering and redirect to a different location with `this.redirect
 	}
 </script>
 ```
+
+#### Typing the function
+
+If you use TypeScript and want to access the above context methods, TypeScript will thow an error and tell you that it does not know the type of `this`. To fix it, you need to specify the type of `this` (see the [official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions)). We provide you with helper interfaces so you can type the function like this:
+
+```html
+<script context="module">
+	import type { PreloadContext, PreloadPage } from "@sapper/common";
+
+	export async function preload(this: PreloadContext, page: PreloadPage, session: any) {
```

Comment:
should we try to add a type for `session` in this example as well?

Reply:
From reading the docs I though this is user-defined. Do you mean something like this?

```ts
import type { PreloadContext, PreloadPage } from "@sapper/common";
import type { Session } from './your-own-session-interface';

export async function preload(this: PreloadContext, page: PreloadPage, session: Session) {
...
```

Reply:
Oh, nevermind then. There's also a `session` store and I thought that there would be a store interface we'd want to use here, but looking at the docs it looks like this might be different. I'm not that familiar with `session` myself yet. I think we can leave this as is